packages/app/src/runner/aut-iframe.ts

```diff
@@ -502,4 +506,292 @@ export class AutIframe {
+          // its inside of our border + padding
+          obj = {
+            width: dimensions.width,
+            height: dimensions.height,
+            top: dimensions.offset.top + dimensions.borderTop + dimensions.paddingTop,
+            left: dimensions.offset.left + dimensions.borderLeft + dimensions.paddingLeft,
+          }
+
+          break
+        default:
+          obj = {
+            width: this._getDimensionsFor(dimensions, attr, 'width'),
+            height: this._getDimensionsFor(dimensions, attr, 'height'),
+            top: dimensions.offset.top,
+            left: dimensions.offset.left,
+          }
+      }
+
+      // if attr is margin then we need to additional
+      // subtract what the actual marginTop + marginLeft
+      // values are, since offset disregards margin completely
+      if (attr === 'Margin') {
+        obj.top -= dimensions.marginTop
+        obj.left -= dimensions.marginLeft
+      }
+
+      if (attr === 'Padding') {
+        obj.top += dimensions.borderTop
+        obj.left += dimensions.borderLeft
+      }
+
+      // bail if the dimensions of this layer match the previous one
+      // so we dont create unnecessary layers
+      if (this._dimensionsMatchPreviousLayer(obj, container)) return
+
+      this._createLayer($el.get(0), attr, color, container, obj)
+    })
+
+    body.appendChild(container)
+
+    for (let i = 0; i < container.children.length; i++) {
+      const el = container.children[i] as HTMLElement
+      const top = parseFloat(el.getAttribute('data-top')!)
+      const left = parseFloat(el.getAttribute('data-left')!)
+
+      setOffset(el, { top, left })
+    }
+
+    return container
+  }
+
+  private _createLayer (el, attr, color, container, dimensions) {
+    const div = document.createElement('div')
+
+    div.style.transform = getComputedStyle(el, null).transform
+    div.style.width = `${dimensions.width}px`
+    div.style.height = `${dimensions.height}px`
+    div.style.position = 'absolute'
+    div.style.zIndex = `${getZIndex(el)}`
+    div.style.backgroundColor = color
+
+    div.setAttribute('data-top', dimensions.top)
+    div.setAttribute('data-left', dimensions.left)
+    div.setAttribute('data-layer', attr)
+
+    container.prepend(div)
+
+    return div
+  }
+
+  private _dimensionsMatchPreviousLayer (obj, container) {
+    // since we're prepending to the container that
+    // means the previous layer is actually the first child element
+    const previousLayer = container.childNodes[0]
+
+    // bail if there is no previous layer
+    if (!previousLayer) {
+      return
+    }
+
+    return obj.width === previousLayer.offsetWidth &&
+      obj.height === previousLayer.offsetHeight
+  }
+
+  private _getDimensionsFor (dimensions, attr, dimension) {
+    return dimensions[`${dimension}With${attr}`]
+  }
+
+  private listeners: any[] = []
+
+  private _addOrUpdateSelectorPlaygroundHighlight ($el, $body, selector?, showTooltip?, onClick?) {
```

Comment:
Any reason you changed this function to be a list of parameters rather than a single options object? I'm a fan of the latter as it makes reading whoever calls this function a lot easier to parse.
```suggestion
  private _addOrUpdateSelectorPlaygroundHighlight ({ $el, $body, selector, showTooltip, onClick }: any) {
```

Reply:
I cannot remember clearly why I changed this. I think I did this because the order of arguments are not consistent and it was hard to copy and paste. 

I rolled it back.